### PR TITLE
Fix compilation without deprecated OpenSSL APIs

### PR DESCRIPTION
**Pull request by neheb**
Leftover stuff from https://github.com/PurpleI2P/i2pd/pull/1278

### CODE REVIEW

File: libi2pd/Crypto.cpp [hunk 1]
```diff
@@ -1246,7 +1246,9 @@ namespace crypto
 	void InitCrypto (bool precomputation)
 	{
 		i2p::cpu::Detect ();	
+#if LEGACY_OPENSSL
```
Comment (orignal) [inline libi2pd/Crypto.cpp]: how do we initialize in case of 1.1?

Comment (neheb) [reply libi2pd/Crypto.cpp]: You don't. 1.1 handles this automatically. See: https://www.openssl.org/news/changelog.html#x12

Comment (r4sas) [reply libi2pd/Crypto.cpp]: That?

 ```
 *) Added support for auto-initialisation and de-initialisation of the library.
     OpenSSL no longer requires explicit init or deinit routines to be called,
     except in certain circumstances. See the OPENSSL_init_crypto() and
     OPENSSL_init_ssl() man pages for further information.
     [Matt Caswell]
```

Comment (neheb) [reply libi2pd/Crypto.cpp]: Yes. OPENSSL_init_ssl only needs to be called in case of non-default options such as initializing the engine.

**Pull request merged**